pkg/ddevapp/craftcms.go

```diff
@@ -111,42 +85,25 @@ func craftCmsPostStartAction(app *DdevApp) error {
 		port = "5432"
 	}
 
-	// If they have older version of .env with DB_DRIVER, DB_SERVER etc, use those
-	if _, ok := envMap["DB_SERVER"]; ok {
-		// TODO: Remove, was never an official standard of Craft CMS.
-		envMap = map[string]string{
-			"DB_DRIVER":             driver,
-			"DB_SERVER":             "db",
-			"DB_PORT":               port,
-			"DB_DATABASE":           "db",
-			"DB_USER":               "db",
-			"DB_PASSWORD":           "db",
-			"MAILPIT_SMTP_HOSTNAME": "127.0.0.1",
-			"MAILPIT_SMTP_PORT":     "1025",
-			"PRIMARY_SITE_URL":      app.GetPrimaryURL(),
-		}
-	} else {
-		// Otherwise use the current CRAFT_DB_SERVER etc.
-		envMap = map[string]string{
-			"CRAFT_DB_DRIVER":       driver,
-			"CRAFT_DB_SERVER":       "db",
-			"CRAFT_DB_PORT":         port,
-			"CRAFT_DB_DATABASE":     "db",
-			"CRAFT_DB_USER":         "db",
-			"CRAFT_DB_PASSWORD":     "db",
-			"CRAFT_WEB_ROOT":        app.GetAbsDocroot(true),
-			"MAILPIT_SMTP_HOSTNAME": "127.0.0.1",
-			"MAILPIT_SMTP_PORT":     "1025",
-			"PRIMARY_SITE_URL":      app.GetPrimaryURL(),
-		}
+	envMap := map[string]string{
+		"CRAFT_DB_DRIVER":       driver,
+		"CRAFT_DB_SERVER":       "db",
+		"CRAFT_DB_PORT":         port,
+		"CRAFT_DB_DATABASE":     "db",
+		"CRAFT_DB_USER":         "db",
+		"CRAFT_DB_PASSWORD":     "db",
+		"CRAFT_WEB_ROOT":        app.GetAbsDocroot(true),
+		"MAILPIT_SMTP_HOSTNAME": "127.0.0.1",
+		"MAILPIT_SMTP_PORT":     "1025",
+		"PRIMARY_SITE_URL":      app.GetPrimaryURL(),
 	}
 
-	err = WriteProjectEnvFile(envFilePath, envMap, envText)
+	err := WriteProjectEnvFile(envFilePath, envMap, "")
```

Comment:
We need to get `ReadProjectEnvFile` back.

Now this code takes full control of the `.ddev/.env.web` file, and you can't set any more env variables there:
```bash
ddev dotenv set .ddev/.env.web --some-variable=foobar
ddev restart
cat .ddev/.env.web # no SOME_VARIABLE
```
But it should be possible.

Reply:
> Now this code takes full control of the .ddev/.env.web file, and you can't set any more env variables there:

Yes, this is the difficulty with not currently having a way to control what's "owned by DDEV" in dotenv files. 

* https://github.com/ddev/ddev/issues/7235 would help with this (I'd use .ddev/.env.global.craftcms probably) but it still wouldn't be authoritative.

Reply:
But it's already working with `ddev dotenv set` (otherwise this command would override the whole file).

This PR should work the same way, it should change only env variables that are related to Craft CMS.

Reply:
Ideally, `WriteProjectEnvFile` should support append or create mode. It's annoying that we need extra code to append or replace env vars in an existing file.

Reply:
Ah, so we could use the logic/function from `ddev dotenv set`??? Awesome!